v2/workloadapi/client.go

```diff
@@ -353,6 +347,49 @@ func defaultClientConfig() clientConfig {
 	}
 }
 
+func parseX509Context(resp *workload.X509SVIDResponse) (*X509Context, error) {
+	svids, err := parseX509SVIDs(resp, -1)
+	if err != nil {
+		return nil, err
+	}
+
+	bundles, err := parseX509Bundles(resp)
+	if err != nil {
+		return nil, err
+	}
+
+	return &X509Context{
+		SVIDs:   svids,
+		Bundles: bundles,
+	}, nil
+}
+
+// parseX509SVIDs parses between 1 and n SVIDs from the response. It fails
+// if it cannot parse at least one, since the SVID is required in the response
+// and the spec indicates clients SHOULD reject such messages. If n is less
+// than one, then all SVIDs are returned.
+func parseX509SVIDs(resp *workload.X509SVIDResponse, n int) ([]*x509svid.SVID, error) {
+	switch {
+	case n < 1, n > len(resp.Svids):
+		n = len(resp.Svids)
+	}
+
+	svids := make([]*x509svid.SVID, 0, n)
+	for i := 0; i < n; i++ {
+		svid := resp.Svids[i]
+		s, err := x509svid.ParseRaw(svid.X509Svid, svid.X509SvidKey)
+		if err != nil {
+			return nil, err
+		}
+		svids = append(svids, s)
+	}
+
+	if len(svids) == 0 {
+		return nil, errors.New("no SVIDs in response")
+	}
+	return svids, nil
+}
+
 func parseX509Bundles(resp *workload.X509SVIDResponse) (*x509bundle.Set, error) {
```

Comment:
Should we also fail if `resp.Svids` is empty?

Reply:
This comes back to my question in the PR description on whether or not we fail when the response has unexpected values even when the function call isn't strictly consuming that part of the response.

The parseX509SVIDs method does ensure that there is at least one SVID and that is used when we want to return X509-SVIDs.

Reply:
Oh! It is ok, I didn't realize that parseX509SVIDs was checking that.